oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsUsbDevice.java

```diff
@@ -55,269 +57,256 @@
-            return device;
-        }
-        return null;
-    }
+	/**
+	 * Recursively creates WindowsUsbDevices by fetching information from maps to
+	 * populate fields
+	 *
+	 * @param hubDeviceId    The PNPdeviceID of this device.
+	 * @param vid            The default (parent) vendor ID
+	 * @param pid            The default (parent) product ID
+	 * @param deviceTreeMap
+	 * @param usbDeviceCache
+	 * @return A WindowsUsbDevice corresponding to this deviceID, or null if unable
+	 *         to find
+	 */
+	private static WindowsUsbDevice getDeviceAndChildren(String hubDeviceId, String vid, String pid,
+			Map<String, List<String>> deviceTreeMap, Map<String, WindowsUsbDevice> usbDeviceCache) {
+		String vendorId = vid;
+		String productId = pid;
+		Matcher m = VENDOR_PRODUCT_ID.matcher(hubDeviceId);
+		if (m.matches()) {
+			vendorId = m.group(1).toLowerCase();
+			productId = m.group(2).toLowerCase();
+		}
+		List<String> pnpDeviceIds = deviceTreeMap.getOrDefault(hubDeviceId, new ArrayList<String>());
+		List<WindowsUsbDevice> usbDevices = new ArrayList<>();
+		for (String pnpDeviceId : pnpDeviceIds) {
+			WindowsUsbDevice deviceAndChildren = getDeviceAndChildren(pnpDeviceId, vendorId, productId, deviceTreeMap,
+					usbDeviceCache);
+			if (deviceAndChildren != null) {
+				usbDevices.add(deviceAndChildren);
+			}
+		}
+		Collections.sort(usbDevices);
+		if (usbDeviceCache.containsKey(hubDeviceId)) {
+			WindowsUsbDevice device = usbDeviceCache.get(hubDeviceId);
+			if (device.name.isEmpty()) {
+				device.name = vendorId + ":" + productId;
+			}
+			device.vendorId = vendorId;
+			device.productId = productId;
+			device.connectedDevices = usbDevices.toArray(new WindowsUsbDevice[0]);
+			return device;
+		}
+		return null;
+	}
 
-    /**
-     * Queries the USB Controller list, which doesn't change so we only need to
-     * query it once
-     *
-     * @param wmiQueryHandler
-     *
-     * @return A list of Strings of USB Controller PNPDeviceIDs
-     */
-    private static List<String> getControllerDeviceIdList(WmiQueryHandler wmiQueryHandler) {
-        if (controllerDeviceIds == null) {
-            controllerDeviceIds = new ArrayList<>();
-            // One time lookup of USB Controller PnP Device IDs which don't
-            // change
-            WmiQuery<USBControllerProperty> usbControllerQuery = new WmiQuery<>("Win32_USBController",
-                    USBControllerProperty.class);
-            WmiResult<USBControllerProperty> usbController = wmiQueryHandler.queryWMI(usbControllerQuery);
-            for (int i = 0; i < usbController.getResultCount(); i++) {
-                controllerDeviceIds.add(WmiUtil.getString(usbController, USBControllerProperty.PNPDEVICEID, i));
-            }
-        }
-        return controllerDeviceIds;
-    }
+	/**
+	 * Queries the USB Controller list, which doesn't change so we only need to
+	 * query it once
+	 *
+	 * @param wmiQueryHandler
```

Comment:
Since you've deleted the method parameter, you should also delete the `@param` javadoc.